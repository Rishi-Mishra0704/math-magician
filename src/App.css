.App {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: flex-start;
}

h1 {
  font-size: 4em;
  font-weight: 700;
  margin: 10px 20px;
  padding: 0;
}

.calculator {
  position: fixed;
  right: 250px;
  bottom: 50px;
}

.quote {
  position: absolute;
  top: 200px;
  left: 20px;
  font-size: 28px;
  font-weight: 600;
}
