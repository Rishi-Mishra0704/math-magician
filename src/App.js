import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import HomePage from './components/Homepage/HomePage';
import Calculator from './components/calculator/Calculator';
import Root from './components/Root';
import QuotesPage from './components/quotes/QuotesPage';

const App = () => (
  <BrowserRouter>
    <Routes>
      <Route path="/" element={<Root />}>
        <Route path="/" element={<HomePage />} />
        <Route path="/calculator" element={<Calculator />} />
        <Route path="/quotes" element={<QuotesPage />} />
      </Route>
    </Routes>
  </BrowserRouter>
);
export default App;
