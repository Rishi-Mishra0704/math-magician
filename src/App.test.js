import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('App', () => {
  test('renders homepage by default', () => {
    const { container } = render(<App />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders calculator page when clicked on the link', () => {
    const { container } = render(<App />);
    fireEvent.click(screen.getByText('Calculator'));
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders quotes page when clicked on the link', () => {
    const { container } = render(<App />);
    fireEvent.click(screen.getByText('Quotes'));
    expect(container.firstChild).toMatchSnapshot();
  });
});
