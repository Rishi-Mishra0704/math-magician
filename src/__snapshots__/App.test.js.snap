// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders calculator page when clicked on the link 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`App renders homepage by default 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;

exports[`App renders quotes page when clicked on the link 1`] = `
<header>
  <nav
    class="navbar"
  >
    <h1>
      Math Magician
    </h1>
    <ul
      class="navList"
    >
      <li>
        <a
          class="navLink"
          href="/"
        >
          HomePage
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink-last-child"
          href="/quotes"
        >
          Quotes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
